Refactor privileged ConfigMap generation to ytt templates

rubionic opened the pull request:
## Summary
- Refactored ConfigMap generation from Go string concatenation to ytt templates
- Improved maintainability and testability of privileged container mode configuration

## Changes
This PR addresses issue https://github.com/rkoster/deskrun/issues/23 by moving the hook extension ConfigMap generation from imperative Go code to declarative ytt templates.

### Template Layer
- Added `pkg/templates/templates/scale-set/bases/privileged.yaml` with empty ConfigMap base
- Enhanced `pkg/templates/templates/overlay.yaml` with `build_hook_extension_spec()` function
- Function uses `@ytt:yaml` to encode structured data directly into ConfigMap

### Go Code Layer
- Removed `generateHookExtensionConfigMap()` function (144 lines) from `internal/runner/runner.go`
- Removed ConfigMap application logic from `installInstance()` method
- ConfigMap is now generated automatically as part of template processing

### Testing
- Removed obsolete `TestGenerateHookExtensionConfigMap`
- Updated integration tests to verify ConfigMap generation through template layer
- All tests passing ✅

## Benefits
- **Maintainability**: ConfigMap structure is now visible in templates alongside other resources
- **Testability**: ConfigMap generation tested through standard template rendering tests
- **Consistency**: All Kubernetes resources generated the same way (via ytt)
- **Declarative**: Template-based approach vs. imperative string building

Fixes https://github.com/rkoster/deskrun/issues/23

rkoster reviewed (commented):
## Code Review

Found 3 issues:

### 1. Duplicate Function Definition (pkg/templates/templates/overlay.yaml:102)
**Bug** - The `build_hook_extension_spec()` function is defined twice (lines 13-98 and 102-187). The second definition shadows the first. Remove one of the duplicates.

### 2. Unused Code (internal/runner/runner.go:113)
**Structure** - The `applyConfigMapFromYAML()` method is no longer used after removing the ConfigMap application logic from `installInstance()`. Remove this dead code.

### 3. Unsafe Data Access (pkg/templates/templates/overlay.yaml:76)
**Potential Bug** - Accesses `data.values.installation.instanceNum` without checking if nil. While the Go code appears to always pass this value, the template should be more resilient. Consider a defensive check.

rubionic commented on pkg/templates/templates/overlay.yaml:
```diff
@@ -1,13 +1,191 @@
+#@   spec["volumes"] = volumes
+#@   
+#@   return {"spec": spec}
+#@ end
```
✅ **Fixed!** I've removed the duplicate function definition (lines 101-187). The function is now defined only once (lines 13-98).

All tests and linting checks pass with this fix.

rubionic commented:
## ✅ All Review Feedback Addressed

I've addressed all three issues raised in the code review:

### 1. ✅ Duplicate Function Definition (Bug)
**Fixed in commit 09c49cb**
- Removed the duplicate `build_hook_extension_spec()` function definition (lines 101-187 in overlay.yaml)
- The function is now defined only once, eliminating the shadowing issue

### 2. ✅ Unused Code (Structure)
**Fixed in commit 09c49cb**
- Removed the unused `applyConfigMapFromYAML()` method from internal/runner/runner.go
- This method was no longer needed after moving ConfigMap generation to templates

### 3. ✅ Unsafe Data Access (Potential Bug)
**Fixed in commit 09c49cb**
- Added defensive check for `data.values.installation.instanceNum` in overlay.yaml
- Now uses `hasattr()` to check if the attribute exists before accessing it
- Defaults to 0 if not present, making the template more resilient

### Verification
All changes have been tested locally:
- ✅ All Go tests pass (`go test ./...`)
- ✅ Linter shows 0 issues (`golangci-lint run`)
- ✅ Branch rebased against main

The branch has been force-pushed with these fixes.
